pokemongo_bot/cell_workers/evolve_all_worker.py

```diff
@@ -20,34 +21,220 @@ def work(self):
+
+            ## at this point release_cand_list contains evolved pokemons data
+            for cand in release_cand_list:
+                pokemon_id = cand[0]
+                pokemon_name = cand[1]
+                pokemon_cp = cand[2]
+                pokemon_potential = cand[3]
+
+                if self.should_release_pokemon(pokemon_name, pokemon_cp, pokemon_potential):
+                    # Transfering Pokemon
+                    self.transfer_pokemon(pokemon_id)
+                    logger.log(
+                        '[#] {} has been exchanged for candy!'.format(pokemon_name), 'green')
+
+    def _sort_by_cp(self, inventory_items):
+        pokemons = []
+        for item in inventory_items:
+            try:
+                reduce(dict.__getitem__, [
+                       "inventory_item_data", "pokemon_data"], item)
+            except KeyError:
+                pass
+            else:
+                try:
+                    pokemon = item['inventory_item_data']['pokemon_data']
+                    pokemon_num = int(pokemon['pokemon_id']) - 1
+                    pokemon_name = self.bot.pokemon_list[int(pokemon_num)]['Name']
+                    pokemons.append([
+                        pokemon['id'],
+                        pokemon_name,
+                        pokemon['cp'],
+                        self._compute_iv(pokemon)
+                        ])
+                except:
                     pass
-                else:
-                    try:
-                        pokemon = item['inventory_item_data']['pokemon_data']
-                        self._execute_pokemon_evolve(pokemon, cache)
-                        sleep(1.2)
-                    except:
-                        pass
+
+        pokemons.sort(key=lambda x: x[2], reverse=True)
+        return pokemons
 
     def _execute_pokemon_evolve(self, pokemon, cache):
-        pokemon_num = int(pokemon['pokemon_id']) - 1
-        pokemon_name = self.bot.pokemon_list[
-            int(pokemon_num)]['Name']
+        pokemon_id = pokemon[0]
+        pokemon_name = pokemon[1]
+        pokemon_cp = pokemon[2]
+
         if pokemon_name in cache:
             return
 
-        self.api.evolve_pokemon(pokemon_id=pokemon['id'])
+        self.api.evolve_pokemon(pokemon_id=pokemon_id)
         response_dict = self.api.call()
         status = response_dict['responses']['EVOLVE_POKEMON']['result']
         if status == 1:
-            print('[#] Successfully evolved {}!'.format(
-                pokemon_name
+            print('[#] Successfully evolved {} with {} cp!'.format(
+                pokemon_name, pokemon_cp
             ))
         else:
             # cache pokemons we can't evolve. Less server calls
             cache[pokemon_name] = 1
+        sleep(5.7)
+
+    # TODO: move to utils. These methods are shared with other workers.
+    def transfer_pokemon(self, pid):
```

Comment:
Can you move these out before this PR? It doesn't make sense to have duplicated code between the workers and this is 120 lines of copy and paste.